src/egui_node.rs

```diff
@@ -171,9 +175,9 @@ pub struct EguiNode {
 
 impl EguiNode {
     /// Constructs Egui render node.
-    pub fn new(window_id: WindowId) -> Self {
+    pub fn new(window_entity: Entity) -> Self {
```

Comment:
I tried to update to bevy main before seeing your PR ; and I did a `type WindowEntity = Entity`, to help me understand where the various `Entity` references were supposed to reference a window.

Would that be interesting ? It might be discussed in another issue, upgrading can already be a big enough scope.

Reply:
That could potentially help make things a bit more clear, though I wonder if we should really be storing the entity all over the place vs turning most things into components on the window entity.

Reply:
I'd like us to try the components approach. Not sure what that would look like exactly, but since windows are entities now, storing additional data as components sounds more idiomatic